TopmedPipeline/tests/testthat/test_getAssoc.R

```
@@ -388,5 +461,175 @@ test_that("remove conditional", {
+
+  tmp <- assoc %>%
+    group_by(chr) %>%
+    group_split()
+  lapply(seq_along(tmp), function(x) {
+    ids = tmp[[x]]$id
+    chr = unique(tmp[[x]]$chr) %>% as.character()
+    save(ids, file = insertChromString(file_pattern, chr))
+  })
+  out <- assocFilterByFile(assoc, file_pattern)
+  expect_equal(out, assoc)
+
+  # Extra var file
+  extra_ids <- max(assoc$id) + 1
+  extra_chr <- as.integer(as.character(max(assoc$chr))) + 1
+  save(extra_ids, file=insertChromString(file_pattern, extra_chr))
+  out <- assocFilterByFile(assoc, file_pattern)
+  expect_equal(out$id, assoc$id)
+  expect_equal(names(out), names(assoc))
+  unlink(insertChromString(file_pattern, extra_chr))
+
+  # Some ids selected.
+  tmp <- assoc %>%
+    group_by(chr) %>%
+    dplyr::slice(1:2) %>%
+    group_split()
+  lapply(seq_along(tmp), function(x) {
+    ids = tmp[[x]]$id
+    chr = unique(tmp[[x]]$chr) %>% as.character()
+    save(ids, file = insertChromString(file_pattern, chr))
+  })
+
+  out <- assocFilterByFile(assoc, file_pattern)
+  expect_equal(out$id, tmp %>% bind_rows() %>% pull(id))
+  expect_equal(names(out), names(assoc))
+
+  # Only one chromosome in assoc but varfiles for other chromosomes exist.
+  out <- assocFilterByFile(tmp[[1]], file_pattern)
+  expect_equal(out$id, tmp[[1]]$id)
+  expect_equal(names(out), names(assoc))
+
+  # Assoc has fewer ids than in file across all chromosomes
+  assoc_subset <- assoc %>%
+    dplyr::group_by(chr) %>%
+    dplyr::slice(1) %>%
+    dplyr::ungroup()
+  out <- assocFilterByFile(assoc_subset, file_pattern)
+  expect_equal(out$id, assoc_subset$id)
+  expect_equal(names(out), names(assoc))
+
+  # Only chr1 varfile exists.
+  unlink(insertChromString(file_pattern, 2))
+  out <- assocFilterByFile(assoc, file_pattern)
+  expect_equal(out$id, assoc %>% filter(chr == 1) %>% dplyr::slice(1:2) %>% pull(id))
+  expect_equal(names(out), names(assoc))
+  fail("Should there be a warning if a varfile is missing?")
```

I wasn't sure what behavior we would want here, so I made the test fail with a question. This is covering following situation:
* the user specified a file pattern with a space (e.g., `var_include_chr .RData`)
* the association test results have multiple chromosomes (e.g., 1, 2, and 3)
* one of the expected variant include files does not exist (e.g., `var_include_chr3.RData`).

This could happen if you didn't want to include any results from one chromosome. I'm leaning towards either printing a warning or silently ignoring it. Suggestions?

---

I ended up adding a warning.